pkg/morph/deploy/notary.go

```diff
@@ -925,56 +931,14 @@ func initNotaryDepositDeficiencyHandler(ctx context.Context, l *zap.Logger, b Bl
 		// If we encounter deposit expiration and current Notary balance >=20% of single
 		// transfer, we just increase the expiration time of the deposit, otherwise, we
 		// make transfer.
-
-		const (
-			// GAS:Notary proportion, see scheme above
-			gasProportion = 50
-			// even there is no lack of GAS at the moment, when the balance falls below 1/5
-			// of the supported value - replenish
-			refillProportion = 5
-			// for simplicity, we just make Notary deposit "infinite" not to prolong
-			till = math.MaxUint32
-		)
-
-		if !lackOfGAS { // deposit expired
-			if new(big.Int).Mul(notaryBalance, big.NewInt(refillProportion)).Cmp(singleNotaryDepositAmount) >= 0 {
-				if expirationTxMonitor.isPending() {
-					l.Info("previously sent transaction increasing expiration time of the Notary deposit is still pending, will wait for the outcome")
-					return
-				}
-
-				l.Info("sending new transaction increasing expiration time of the Notary deposit...", zap.Uint32("till", till))
-
-				txID, vub, err := notaryContract.LockDepositUntil(localAccID, till)
-				if err != nil {
-					l.Error("failed to send transaction increasing expiration time of the Notary deposit, will try again later", zap.Error(err))
-					return
-				}
-
-				l.Info("transaction increasing expiration time of the Notary deposit has been successfully sent, will wait for the outcome",
-					zap.Stringer("tx", txID), zap.Uint32("vub", vub))
-
-				expirationTxMonitor.trackPendingTransactionsAsync(ctx, vub, txID)
-
-				return
-			}
-		}
-
-		if transferTxMonitor.isPending() {
-			l.Info("previously sent transaction local account's GAS to the Notary contract is still pending, will wait for the outcome")
-			return
-		}
-
-		needAtLeast := new(big.Int).Mul(singleNotaryDepositAmount, big.NewInt(gasProportion))
-		if gasBalance.Cmp(needAtLeast) < 0 {
-			l.Info("minimum threshold for GAS transfer from local account to the Notary contract not reached, will wait for replenishment",
-				zap.Stringer("need at least", needAtLeast), zap.Stringer("have", gasBalance))
-			return
+		if txMonitor.isPending() {
+			l.Info("previously sent transaction transferring local account's GAS to the Notary contract is still pending, will wait for the outcome")
+			continue
 		}
 
 		var transferData notary.OnNEP17PaymentData
 		transferData.Account = &localAccID
-		transferData.Till = till
+		transferData.Till = math.MaxUint32 // deposit "forever" so we don't have to renew
```

Comment:
@AnnaShaleva, can we have problems in future work after such a deposit? e.g. when that acc will try to deposit the lower GAS amount?

Reply:
> lower GAS amount

lower that _what_ do u mean? afaik any additional deposits will add GAS to the balance and that's it

i moved to `math.MaxUint32` by @AnnaShaleva advice btw https://github.com/nspcc-dev/neofs-node/pull/2408#discussion_r1252029371

Reply:
i meant that we may have 50GAS deposit until maxInt an we do 1GAS deposit until maxInt, does it make the actual deposit 51GAS? i think yes but i always forget some details about notary so i wanted to be sure

if it is ok, then it is ok

Reply:
Yes, it works as you described, and it's OK.

Reply:
The only problem is that this GAS is locked forever. But who cares about this GAS anyway...